### PR TITLE
Add URI Namespace for File & HDFS CSV parsing

### PR DESCRIPTION
**Pull request by ryankwagner**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

Work included is tested in local HDFS using the hdfs start-all.sh script (via hadoop docs) on HDFS 3.3.1
File used is a world cities CSV both unzipped and zipped.
Updates the RowCSVWriter to use the 4.6 version of openCSV.
Updates file with significant Druid code usage with Apache license header.

Data parsers can later be updated to include TSV, JSON, ObjectMapper, etc via:
https://www.mvndoc.com/c/io.druid.extensions/druid-lookups-cached-global/io/druid/query/lookup/namespace/class-use/UriExtractionNamespace.FlatDataParser.html

Code usage (taken from old Druid revision):
https://github.com/apache/druid/blob/master/extensions-core/lookups-cached-global/src/main/java/org/apache/druid/query/lookup/namespace/UriExtractionNamespace.java#L277

### CODE REVIEW

File: druid-lookups/pom.xml [hunk 1]
```diff
@@ -178,6 +178,12 @@
             <version>${druid.version}</version>
             <scope>provided</scope>
         </dependency>
+        <dependency>
```
Comment (pranavbhole) [inline druid-lookups/pom.xml:181]: check if is needed

File: druid-lookups/pom.xml [hunk 2]
```diff
@@ -294,6 +300,12 @@
             <type>test-jar</type>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.druid.extensions</groupId>
+            <artifactId>druid-hdfs-storage</artifactId>
+            <version>0.21.1</version>
```
Comment (pranavbhole) [inline druid-lookups/pom.xml:306]: variable

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/URILookupExtractor.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import com.yahoo.maha.maha_druid_lookups.query.lookup.namespace.URIExtractionNamespace;
+import com.yahoo.maha.maha_druid_lookups.server.lookup.namespace.LookupService;
+import org.apache.druid.java.util.common.logger.Logger;
+
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+import java.util.Map;
+
+public class URILookupExtractor<U extends List<String>> extends OnlineDatastoreLookupExtractor<U> {
+    private static final Logger LOG = new Logger(MethodHandles.lookup().lookupClass());
```
Comment (pranavbhole) [inline druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/URILookupExtractor.java:32]: current class name

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/URIExtractionNamespaceCacheFactory.java [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+                                uri.toString(), id, fmt.print(lastModified), fmt.print(lastCheck));
+                        return version;
+                    }
+                } catch (NumberFormatException ex) {
+                    log.debug(ex, "Failed to get last modified timestamp. Assuming no timestamp");
+                }
+                final ByteSource source;
+                if (CompressionUtils.isGz(uriPath)) {
+                    // Simple gzip stream
+                    log.debug("Loading gz");
```
Comment (ryankwagner) [inline druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/URIExtractionNamespaceCacheFactory.java:143]: Test on GZ

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/URILookupExtractor.java (hunk 3 above)
Comment (ryankwagner) [reply druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/URILookupExtractor.java:32]: Consistent with our other extractor loggers:
```
public class JDBCLookupExtractor<U extends List<String>> extends OnlineDatastoreLookupExtractor<U> {
    private static final Logger LOG = new Logger(MethodHandles.lookup().lookupClass());
    
    public class MongoLookupExtractor<U extends List<String>> extends OnlineDatastoreLookupExtractor {
    private static final Logger LOG = new Logger(MethodHandles.lookup().lookupClass());
```

File: druid-lookups/pom.xml (hunk 1 above)
Comment (ryankwagner) [reply druid-lookups/pom.xml:181]: Removing from lookups, bumping version & fixing in core.

**Review by pranavbhole — APPROVED** [review]

**Pull request merged**